messages/java/src/main/java/io/cucumber/messages/Messages.java

```diff
@@ -2284,7 +2284,7 @@ public ParameterType(
             String id
         ) {
             this.name = name;
-            this.regularExpressions = regularExpressions;
+            this.regularExpressions = regularExpressions == null ? null : java.util.Collections.unmodifiableList(regularExpressions);
```

Comment:
Consider using `Collections.emptyList()` instead of `null`. 

Also can there be a parameter type without a regular expression?

Reply:
Also consider using `Collections.unmodifiableList(new ArrayListt(regularExpressions))` to ensure that the the collection in `ParameterType` can not be modffified by accident by any one who still holds a reference to `regularExpressions`.

Reply:
What the! I always thought `unmodifiableList` made a copy rather than holding a reference. But it makes sense - someone might want to get updates made elsewhere without being allowed to update it. Fixed.

Reply:
> Consider using `Collections.emptyList()` instead of `null`.
> 
> Also can there be a parameter type without a regular expression?

The `ParameterType.regularExpressions` field is not nullable. This is verified in `validate()`. 

I added a few [minItems])(https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.4.2) in 29f85f7380c37d511bdbec1b54ede5e75597b31c and updated the code generator to verify the number of elements.

Reply:
> Consider using `Collections.emptyList()` instead of `null`

I don't think we should override what the user passes in. If they want an empty list, pass an empty list. Passing null when a list is required should be an error (validated later in `validate()`)